packages/wrangler/src/inspect.ts

```diff
@@ -296,88 +437,113 @@ export default function useInspector(props: InspectorProps) {
-													// > To save space, the original name is only recorded when it's different from the final name.
-													// however, source-map consumer does not handle this
-													if (pos && pos.line != null) {
-														const convertedFnName =
-															pos.name || functionName || "";
-														exceptionLines.push(
-															`    at ${convertedFnName} (${pos.source}:${pos.line}:${pos.column})`
-														);
-													}
-												}
-											} catch {
-												// Line failed to parse through the sourcemap consumer
-												// We should handle this better
-											}
-										}
-									);
-								}
+							const message =
+								params.exceptionDetails.exception?.description?.split("\n")[0];
+							const stack = params.exceptionDetails.stackTrace?.callFrames;
+							const formatted = formatStructuredError(
+								sourceMapConsumer,
+								message,
+								stack
 							);
-
 							// Log the parsed stacktrace
-							logger.error(
-								params.exceptionDetails.text,
-								exceptionLines.join("\n")
-							);
+							logger.error(params.exceptionDetails.text, formatted);
 						} else {
 							// We log the stacktrace to the terminal
 							logger.error(
 								params.exceptionDetails.text,
 								params.exceptionDetails.exception?.description ?? ""
 							);
 						}
+
+						// Fetch additional properties if the error has a `cause` property,
+						// or it's a `DOMException`. The `preview` may actually contain the
+						// full `cause` stack trace, but it's likely truncated.
+						const hasCause =
+							params.exceptionDetails.exception?.preview?.properties.some(
+								(prop) => prop.name === "cause"
+							);
+						// `DOMException`s are special, the actually useful stack trace
+						// is hidden behind the `stack` getter, and not included in the
+						// preview: https://github.com/cloudflare/workerd/blob/1b5057f2bfcfedf146f6f79ff04e99903d55412b/src/workerd/jsg/dom-exception.h#L92
+						const isDomException =
+							params.exceptionDetails.exception?.className === "DOMException";
+
+						// The two requests here are based on intercepted messages sent by
+						// the DevTools frontend to view the stack.
+						const objectId = params.exceptionDetails.exception?.objectId;
+						if (objectId !== undefined && (hasCause || isDomException)) {
+							// Get all properties
+							const getPropertiesResponse = await send<
+								Protocol.Runtime.GetPropertiesRequest,
+								Protocol.Runtime.GetPropertiesResponse
+							>("Runtime.getProperties", {
+								objectId,
+								ownProperties: false,
+								accessorPropertiesOnly: false,
+								generatePreview: false,
+								nonIndexedPropertiesOnly: false,
+							});
+
+							// If this error has a `cause` property, display it
+							const causeDescriptor = getPropertiesResponse?.result.find(
+								(prop) => prop.name === "cause"
+							);
+							const cause = causeDescriptor?.value;
```

Comment:
AFAICT [the spec](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause#providing_structured_data_as_the_error_cause) for `Error.cause` doesn't define a structure for the cause property -  where's`cause.description` and `cause.value` etc coming from?